streams/src/main/java/org/apache/kafka/streams/state/internals/SegmentIterator.java

```diff
@@ -82,9 +93,9 @@ public boolean hasNext() {
                     }
                 } else {
                     if (forward) {
-                        currentIterator = currentSegment.range(from, to);
+                        currentIterator = currentSegment.range(from, to, prefixScan);
```

Comment:
Instead of passing this boolean around and creating overloaded interfaces, could we:

* add `prefixScan` in Segment / SegmentIterator.
* add a different `RocksDBPrefixIterator` besides `RocksDBRangeIterator` which would be used in `prefixScan`, and then the `RocksDBStore#prefixScan` to be used in SegmentIterator.

Reply:
I should clarify: not meant the existing `RocksDBStore#prefixScan` since it only allows a single prefix, but adding another function.